### PR TITLE
Dropdown

### PR DESCRIPTION
**Pull request by lekowan**
Created a dropdown menu using Radix. The dropdown menu pulls data from a language-list.json.

### CODE REVIEW

File: src/data/language/language.ts
Comment (seanmturley) [inline src/data/language/language.ts:1]: I think `getLanguageList` could be simpler:

```
export const getLanguageList = async () => {
  try {
    return await import(`@data/language-list.json`)
  } catch (_error) {
    throw new Error(`Failed to load language list`)
  }
}
```

File: src/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -1,20 +1,37 @@
-import { usePathname, useRouter } from "next/navigation"
+import { useParams, usePathname, useRouter } from "next/navigation"
+import * as DropdownUI from "@radix-ui/react-dropdown-menu"
+import { useState } from "react"
+import { LanguageList } from "@data/language"
 
-export const Dropdown = () => {
-  const language = usePathname()
+export const Dropdown = ({ languageList }: { languageList: LanguageList }) => {
```
Comment (seanmturley) [inline src/components/dropdown/dropdown.tsx:6]: I'm suspicious of the prop drilling required to get `languageList` here. It's passed down through 4 levels: 

`/[language]/page.tsx` > `StoryListTemplate` > `Header` > `Dropdown`

None of the levels above `Dropdown` actually use the data.

Do you want to go with it for now, and see if we can refine it later?

File: src/components/dropdown/dropdown.tsx [hunk 2]
```diff
@@ -1,20 +1,37 @@
...
-      <option value="/fr">French</option>
-    </select>
+    <DropdownUI.Root>
+      <DropdownUI.Trigger>
+        {languageList[`${language}`].name}
+      </DropdownUI.Trigger>
+      <DropdownUI.Content className="tw-bg-white tw-p-1 tw-border tw-border-gray-400">
+        {Object.keys(languageList).map((language: string) => (
+          <DropdownUI.Item
+            key="language"
```
Comment (seanmturley) [inline src/components/dropdown/dropdown.tsx:27]: The `key` of each child element needs to be unique (there are console errors in the browser). I think you probably meant for this to be:
```
key={language}
```

File: src/components/dropdown/dropdown.tsx [hunk 3]
```diff
@@ -1,20 +1,37 @@
...
   const router = useRouter()
+  const [language, setLanguage] = useState(languageParam)
 
-  const handleLanguageChange = (
-    event: React.ChangeEvent<HTMLSelectElement>
-  ) => {
-    const selectedLanguage = event.target.value
-    router.push(`${selectedLanguage}`)
+  const handleLanguageChange = (language: string) => {
+    const newPath = pathname.replace(/\/(fr|jp)(\/|$)/, `/${language}/`)
```
Comment (seanmturley) [inline src/components/dropdown/dropdown.tsx:14]: Is the goal of this just to navigate between `/fr` and `/jp` and vice versa?

It works, so do you want to go with it for now? We could simplify it later.

**Review by seanmturley — CHANGES_REQUESTED** [review]
It looks like this is working well.

I suggested a few small changes to make some things clearer or simpler, and to get rid of some console logging and errors.

I've pointed out the potential for some bigger refinements to be made, but maybe we look into those afterwards.

File: src/app/[language]/page.tsx [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import { StoryListTemplate } from "@components/story-list-template"
+import { getLanguageList } from "@data/language/language"
 import { getStoriesList } from "@data/story"
 
 export async function generateStaticParams() {
-  const languages = ["jp", "fr"]
-  return languages.map((language) => ({ language }))
+  const languageList = await getLanguageList()
+  console.log(Object.keys(languageList))
```
[the remark this answers is not in the document]
Comment (lekowan) [reply src/app/[language]/page.tsx:7]: Yep. Done.

File: src/components/header/header.tsx [hunk 5]
```diff
@@ -1,11 +1,12 @@
 import { Dropdown } from "@components/dropdown"
 import { Logo } from "@components/logo"
+import { LanguageList } from "@data/language/language"
```
[the remark this answers is not in the document]
Comment (lekowan) [reply src/components/header/header.tsx:3]: Good catch

File: src/components/dropdown/dropdown.tsx (hunk 1 above)
Comment (lekowan) [reply src/components/dropdown/dropdown.tsx:6]: Yes. Let's make a note of it. I had started to put it in a zustand store but couldn't get it to work (I was on the train then so not the best environment) so went with the easy option. I will make a ticket for refactoring this, feel free to pick it up.

File: src/components/dropdown/dropdown.tsx (hunk 2 above)
Comment (lekowan) [reply src/components/dropdown/dropdown.tsx]: Yes typo

File: src/components/dropdown/dropdown.tsx (hunk 3 above)
Comment (lekowan) [reply src/components/dropdown/dropdown.tsx:14]: That's right. This replace isn't pretty. Yes, let's find something more elegant later

File: src/components/dropdown/dropdown.tsx [hunk 6]
```diff
@@ -1,20 +1,37 @@
-import { usePathname, useRouter } from "next/navigation"
+import { useParams, usePathname, useRouter } from "next/navigation"
+import * as DropdownUI from "@radix-ui/react-dropdown-menu"
+import { useState } from "react"
+import { LanguageList } from "@data/language"
```
[the remark this answers is not in the document]
Comment (lekowan) [reply src/components/dropdown/dropdown.tsx]: Good point

**Review by seanmturley — APPROVED** [review]

**Pull request merged**